docs: Describe current issue w/CRA 5 integration (#2010)
mdx-js/mdx: docs/docs/getting-started.server.mdx

through webpack loader syntax in imports.

Install the webpack loader [`@mdx-js/loader`][mdx-loader].

> To enable direct MDX imports w/o `!@mdx-js/loader!` prefix, the loader can

Reviewer: ```suggestion
To enable MDX imports without `!@mdx-js/loader!` prefix, the loader can
```

> further be added to webpack’s config, using a

Reviewer: ```suggestion
be added to the webpack config, using a
```

> [`react-scripts`](http://github.com/facebook/create-react-app/tree/main/packages/react-scripts)

Reviewer: ```suggestion
`react-scripts`
```

I don’t think this has to be linked, as it does not add something useful to users that are trying to solve this problem, to open it in a tab?

> rewiring tool, e.g. [CRACO](http://github.com/gsoft-inc/craco).

Reviewer: ```suggestion
rewiring tool, such as [CRACO](http://github.com/gsoft-inc/craco).
```

`such as`, compared to a latin abbreviation, is often easier to understand.

Are the popular and viable others?

Author: > Are the popular and viable others?

Popular, yes: https://github.com/timarney/react-app-rewired – the standard for years! – only not compatible yet w/CRA 5; which might not change, given its poor recent maintenance … So actually: CRACO is the only viable one. I'll further rephrase that section …

<details>
  <summary>Expand CRACO example</summary>

  ```js path="craco.config.js"

>   const { addAfterLoader, loaderByName } = require('@craco/craco')
> 
>   module.exports = {
>     webpack: {
>       configure: (webpackConfig) => {
>         addAfterLoader(webpackConfig, loaderByName('babel-loader'), {
>           test: /\.(md|mdx)$/,
>           loader: require.resolve('@mdx-js/loader')
>         })
> 
>         return webpackConfig
>       }
>     }
>   }

Reviewer: ```suggestion
  const {addAfterLoader, loaderByName} = require('@craco/craco')

  module.exports = {
    webpack: {
      configure(webpackConfig) {
        addAfterLoader(webpackConfig, loaderByName('babel-loader'), {
          test: /\.mdx?$/,
          loader: require.resolve('@mdx-js/loader')
        })
        return webpackConfig
      }
    }
  }
```

- match project code style
- use normal function, there’s no reason to use an arrow function
- use the same regex used in other places

  ```

  ```jsx path="src/App.jsx"
  import Content from './content.mdx'

  export default function App() {
    return <Content />
  }
  ```
</details>

<Note type="info">

>   **Note**: due to [broken MDX import forwarding](http://github.com/facebook/create-react-app/issues/12166)
>   in `react-scripts` 5.x, rewiring is currently necessary when using CRA 5.
>   There’s an [ongoing discussion](http://github.com/mdx-js/mdx/discussions/1870)
>   about this issue.

Reviewer: ```suggestion
  **Note**: rewiring is currently needed for CRA 5,
  due to a bug in `react-scripts`
  ([`facebook/create-react-app#12166`](http://github.com/facebook/create-react-app/issues/12166)),
  which is also tracked at
  [`mdx-js/mdx#1870`](http://github.com/mdx-js/mdx/discussions/1870).
```

How about reworded like this? Puts what it means for users (have to rewire) at the front. Then explains and links to in-depth stuff.
